crates/air-lib/trace-handler/src/merger/call_merger.rs

```diff
@@ -18,66 +18,57 @@ mod utils;
 
 use super::*;
 use crate::TracePos;
-use air_parser::ast::CallOutputValue;
 use utils::*;
 
 const EXPECTED_STATE_NAME: &str = "call";
 
 #[derive(Debug, Clone)]
 pub enum MergerCallResult {
     /// There is no corresponding state in a trace for this call.
-    Empty,
+    NotMet,
+
     /// There was a state in at least one of the contexts. If there were two states in
     /// both contexts, they were successfully merged.
-    CallResult {
-        value: CallResult,
-        trace_pos: TracePos,
-        scheme: PreparationScheme,
-    },
+    Met(MetResult),
 }
 
-pub(crate) fn try_merge_next_state_as_call(
-    data_keeper: &mut DataKeeper,
-    output_value: &CallOutputValue<'_>,
-) -> MergeResult<MergerCallResult> {
+#[derive(Debug, Clone)]
+pub struct MetResult {
+    pub result: CallResult,
+    pub trace_pos: TracePos,
+    pub source: ValueSource,
+}
+
+#[derive(Debug, Clone)]
```

Comment:
I would either add `Copy` here, or remove `Clone` here and above (it compiles!).